R/clustering_functions.R

```diff
@@ -1250,11 +1272,26 @@ predict_Medoids = function(data, MEDOIDS = NULL, distance_metric = 'euclidean',
   structure(
     list(clusters = as.vector(res$clusters) + 1,
          fuzzy_clusters = res$fuzzy_clusters,
-         dissimilarity = res$dissimilarity),
-    class = c("MedoidsClusteR", "cluster medoids silhouette"))
+         dissimilarity = res$dissimilarity,
+         distance_metric = distance_metric),
+    class = c("MedoidsCluster", "cluster medoids silhouette"))
 }
 
 
+#' @rdname predict_Medoids
+#' @param object,newdata,... arguments for the `predict` generic
+#' @export
+predict.MedoidsCluster <- function(object, newdata,
+                                   fuzzy = FALSE, threads = 1, ...) {
+  out <- predict_Medoids(newdata, MEDOIDS = object$medoids,
+                         distance_metric = object$distance_metric,
+                         fuzzy = fuzzy, threads = threads)
+  if (fuzzy)
```

Comment:
in the *predict_Medoids()* function when the user specifies *fuzzy = TRUE* then the function returns 3 objects:

- clusters
- fuzzy_clusters
- dissimilarity

Is your intention to follow the same output format of the *predict()* function as is the case with other algorithms that return either a vector or a matrix (in case of binary classification for instance)?

Reply:
ok, then no need of an extra parameter. I thought initially that the previous predict functions will be deprecated. Thus, it's fine the 'predict.MedoidsCluster()' function.